examples/with-nextjs-and-clerk-auth/src/components/NavigationMenu/NavigationListItem.tsx

```diff
@@ -0,0 +1,53 @@
+'use client';
+
+import { PropsWithChildren, ReactNode } from 'react';
+
+import Link from 'next/link';
+import { usePathname } from 'next/navigation';
+
+import {
+  Box,
+  ListItem,
+  ListItemButton,
+  ListItemButtonProps,
+  ListItemIcon,
+  ListItemText,
+  Typography,
+} from '@mui/material';
+
+type NavigationLinkProps = PropsWithChildren<{
+  endIcon?: ReactNode;
+  href?: string;
+  icon: ReactNode;
+  onClick?: ListItemButtonProps['onClick'];
+}>;
+
+export const NavigationListItem = ({
+  endIcon,
+  children,
+  href,
+  icon,
+  onClick,
+}: NavigationLinkProps) => {
+  const pathname = usePathname();
+  const selected = pathname === href?.replace(window.location.origin, '');
```

Comment:
I was thinking about [a more convenient way](https://nextjs.org/docs/app/api-reference/functions/use-pathname) for the next.js

Reply:
@edcs, `'use client'` does not say that this component is executed only on the client. It says that the code of this component is delivered to the client, instead of raw html markup as in the case of Server Components.

<details><summary>console error log</summary>
<p>
<img width="750" alt="image" src="https://github.com/team-monite/monite-sdk/assets/725645/95967bc1-e279-49b8-bfa3-326ee08ca399">
</p>
</details> 

Please be more careful and check every now and then what is being dumped to the console.